src/frontend/components/DashboardThumbnail/index.spec.tsx

```diff
@@ -0,0 +1,282 @@
+import { cleanup, fireEvent, render, wait } from 'react-testing-library';
+
+jest.mock('../../data/sideEffects/createThumbnail/createThumbnail', () => ({
+  createThumbnail: jest.fn(),
+}));
+
+jest.mock('react-router-dom', () => ({
+  Redirect: ({ push, to }: { push: boolean; to: string }) =>
+    `Redirect push to ${to}.`,
+}));
+
+import { DashboardThumbnail } from '.';
+import { bootstrapStore } from '../../data/bootstrapStore';
+import { createThumbnail } from '../../data/sideEffects/createThumbnail/createThumbnail';
+import { appState } from '../../types/AppData';
+import { uploadState } from '../../types/tracks';
+import { jestMockOf } from '../../utils/types';
+
+const mockCreateThumbnail: jestMockOf<
+  typeof createThumbnail
+> = createThumbnail as any;
+
+describe('<DashboardThumbnail />', () => {
+  afterEach(jest.resetAllMocks);
+  afterEach(cleanup);
+
+  // Disable useless async act warnings
+  // TODO: remove this spy as soon as async act is available
+  beforeAll(() => {
+    jest.spyOn(console, 'error');
+  });
+
+  const video = {
+    description: '',
+    id: '43',
+    is_ready_to_play: true,
+    show_download: true,
+    thumbnail: {
+      active_stamp: 128748302847,
+      id: '42',
+      is_ready_to_display: true,
+      upload_state: uploadState.READY,
+      urls: {
+        144: 'https://example.com/thumbnail/144',
+        240: 'https://example.com/thumbnail/240',
+        480: 'https://example.com/thumbnail/480',
+        720: 'https://example.com/thumbnail/720',
+        1080: 'https://example.com/thumbnail/1080',
+      },
+      video: '43',
+    },
+    timed_text_tracks: [],
+    title: '',
+    upload_state: uploadState.READY,
+    urls: {
+      manifests: {
+        dash: 'https://example.com/dash',
+        hls: 'https://example.com/hls',
+      },
+      mp4: {
+        144: 'https://example.com/mp4/144',
+        240: 'https://example.com/mp4/240',
+        480: 'https://example.com/mp4/480',
+        720: 'https://example.com/mp4/720',
+        1080: 'https://example.com/mp4/1080',
+      },
+      thumbnails: {
+        144: 'https://example.com/default_thumbnail/144',
+        240: 'https://example.com/default_thumbnail/240',
+        480: 'https://example.com/default_thumbnail/480',
+        720: 'https://example.com/default_thumbnail/720',
+        1080: 'https://example.com/default_thumbnail/1080',
+      },
+    },
+  };
+
+  it('displays a thumbnail image when the related Thumbnail object is ready', () => {
+    const { getByAltText, queryByText } = render(
+      <Provider
+        store={bootstrapStore({
+          jwt: '',
+          resourceLinkid: '',
+          state: appState.INSTRUCTOR,
+          video,
+        })}
+      >
+        <DashboardThumbnail video={video} />
+      </Provider>,
+    );
+
+    // The progress indicator, processing message & error message are not shown
+    expect(queryByText('0%')).toEqual(null);
+    expect(
+      queryByText(
```

Comment:
why don't using here `getByText` without expect ? If not fount it will raise an error.

Reply:
good catch!